### PR TITLE
[feature] Add probe for SDXL controlnet models

### PR DESCRIPTION
**Pull request by lstein**
## What type of PR is this? (check all applicable)

- [ ] Refactor
- [X] Feature
- [ ] Bug Fix
- [ ] Optimization
- [ ] Documentation Update
- [ ] Community Node Submission

## Have you discussed this change with the InvokeAI team?
- [X] Yes
- [ ] No, because:

      
## Have you updated all relevant documentation?
- [X] Yes
- [ ] No

## Description

This allows SDXL controlnet safetensors models to be auto probed and installed. However, the diffusers library does not yet know how to convert the SDXL models located at https://huggingface.co/lllyasviel/sd_control_collection, so further work needs to be done to make these models work with InvokeAI.

Also, since we are in transition from model manager 1 to model manager 2, this fix had to be applied in two places.

## Related Tickets & Documents

- Related Issue #
- Closes #

## Merge Plan

Merge when approved.

## Added/updated tests?

- [ ] Yes
- [ ] No : _please replace this line with details on why tests
      have not been included_

### CODE REVIEW

File: invokeai/backend/model_management/model_probe.py [hunk 1]
```diff
@@ -404,19 +404,20 @@ class ControlNetCheckpointProbe(CheckpointProbeBase):
...
-            elif checkpoint[key_name].shape[-1] == 1024:
+            elif width == 1024:
                 return BaseModelType.StableDiffusion2
-            elif self.checkpoint_path and self.helper:
-                return self.helper(self.checkpoint_path)
-        raise InvalidModelException("Unable to determine base type for {self.checkpoint_path}")
-
+            elif width == 2048:
+                return BaseModelType.StableDiffusionXL
+        raise InvalidModelException("{self.model_path}: Unable to determine base type")
```
Comment (RyanJDick) [inline invokeai/backend/model_management/model_probe.py:420]: This should be an f-string, and `self.model_path` does not exist:
```suggestion
        raise InvalidModelException(f"{self.checkpoint_path}: Unable to determine base type")
```

File: invokeai/backend/model_manager/probe.py [hunk 2]
```diff
@@ -440,15 +440,19 @@ class ControlNetCheckpointProbe(CheckpointProbeBase):
...
-            if checkpoint[key_name].shape[-1] == 768:
+            width = checkpoint[key_name].shape[-1]
+            if width == 768:
                 return BaseModelType.StableDiffusion1
-            elif checkpoint[key_name].shape[-1] == 1024:
+            elif width == 1024:
                 return BaseModelType.StableDiffusion2
+            elif width == 2048:
+                return BaseModelType.StableDiffusionXL
         raise InvalidModelConfigException("{self.model_path}: Unable to determine base type")
```
Comment (RyanJDick) [inline invokeai/backend/model_manager/probe.py:456]:
```suggestion
        raise InvalidModelConfigException(f"{self.model_path}: Unable to determine base type")
```

**Review by RyanJDick — CHANGES_REQUESTED** [review]
I tested with a couple of SDXL controlnets:
- [diffusers_xl_canny_full.safetensors](https://huggingface.co/lllyasviel/sd_control_collection/blob/main/diffusers_xl_canny_full.safetensors): Succeeded
- [diffusers_xl_canny_small.safetensors](https://huggingface.co/lllyasviel/sd_control_collection/blob/main/diffusers_xl_canny_small.safetensors): Failed (I assume this was supposed to work?)

**Comment by hipsterusername** [comment]
@lstein - Does this need a re-review?

**Comment by lstein** [comment]
@RyanJDick It should be correctly probing all SDXL controlnet models now.

**Review by hipsterusername — APPROVED** [review]

**Pull request merged**